### ISSUE TITLE
How to config the bodyParser limit ?

### ISSUE THREAD

**Issue by loicbisiere** [balderdashy/sails#3348]
Hi, I have a problem to add some configuration to the bodyParser of sails. I need to add this option if I was in a Express standard configuration with bodyParser :
```
app.use(bodyParser.json({limit: '50mb'}));
app.use(bodyParser.urlencoded({limit: '50mb', extended: true}));
```
How can I add this configuration in the "sails context" (0.11.2) ?

Thank you.

**Comment by TurboCoder** [comment]
I think that you can use this:

Under the configurations in sails (that means in the config folder ), go to the http ( config/http.js ) section and then right after the middleware put this:
```
   middleware: {..},
```
```
bodyParser: (function () {
    var opts = {limit: '50mb'};
    var fn;

    // Default to built-in bodyParser:
    fn = require('skipper');
    return fn(opts);
  }),
```
After you do that, you'll need to run `npm install skipper --save`, to get the dependencies needed for this to work.

*Note*: it looks like right now sails doesn't support the body limit by default and you have to use the skipper module for it.

Hope it helps...

**Comment by loicbisiere** [comment]
Hi, I tried your solution with sails 0.11.0, 0.11.2 and 0.12.0-rc3. It doesn't work for me.
I'm currently working with the 0.11.2.
I added some code to get the request headers content-length.
```
var length = req.headers['content-length'];
console.log(length);
```
When the content-length is 589930 (in my case),some params are missing (or truncated). In another case, I get a smaller json object to POST with a content-length 111065 and it works perfectly.

**Comment by tswaters** [comment]
I don't think this is a sails bug.  

After encountering a similar error -- fields being truncated -- I traced through the code to see where it fails.  In my case, the content length was not the issue -- indeed, if the content length is exceeded, an error will be thrown..... what I was seeing was missing body fields.

So, after being received by body-parser, the form post string data did have all the fields I expected -- but after it is run through the qs module to turn it into an object (in our setup, this appears to be 2.2.2) -- the limits that qs module enforces were causing the issue; specifically, the # of fields restricted to 1000 and a depth of 5.  

https://github.com/hapijs/qs/blob/v2.2.2/lib/parse.js#L8-L13

Unfortunately, the body-parser module that comes with the connect module, which comes with sails does not provide any options to the underlying parser so these defaults will continue to be used.

--

In light of this, we're going to rework our `$.post` calls to `$.ajax` passing the content type as application/json -- this appears to work properly without any of the same limits...  well, nothing that I've hit anyway with the same payload :)

**Comment by TurboCoder** [comment]
As far as i can see here, it may be one of this two options:

* You're not using it in the middleware in the `order`, check if you have the middleware bodyparser that you added
```
order: [
      'startRequestTimer',
      'cookieParser',
      //'session',
      'logger',
      'bodyParser',
      'handleBodyParserError',
      'compress',
      'methodOverride',
      'poweredBy',
      '$custom',
      'router',
      'www',
      'favicon',
      '404',
      '500'
    ],
```

* You need to upgrade your limit to 100mb or more.
```
bodyParser: (function () {
    var opts = {limit: 100mb'};
    var fn;

    // Default to built-in bodyParser:
    fn = require('skipper');
    return fn(opts);
  }),
```

*Note*:  In order to debug i recommend you to use the tools out there from [jetbrains](https://www.jetbrains.com/) or a debug server, remember that you can also print in the middleware `bodyparser` in order to check if you were there before the route in the request.

**Comment by sailsbot** [comment]
Thanks for posting, @loicbisiere.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
